hera_mc/cm_handling.py

```diff
@@ -160,13 +140,11 @@ def find_ports(self, connection_dossier):
 
         input_ports = []
         output_ports = []
-        for k, v in connection_dossier.iteritems():
-            if k in self.non_class_connection_dossier_entries:
-                continue
-            if ':up:' in k and v.downstream_input_port not in input_ports:
-                input_ports.append(v.downstream_input_port)
-            elif ':down:' in k and v.upstream_output_port not in output_ports:
-                output_ports.append(v.upstream_output_port)
+        for k, c in connection_dossier['connections'].iteritems():
+            if c.downstream_input_port not in input_ports:
+                input_ports.append(c.downstream_input_port)
+            elif c.upstream_output_port not in output_ports:
+                output_ports.append(c.upstream_output_port)
```

Comment:
Well, actually, what I had in mind would change the semantics. Is the `elif` in the current code strictly necessary? If not, I think you could write:

```
input_ports = set()
output_ports = set()

for k, c in connection_dossier['connections'].iteritems():
    input_ports.add(c.downstream_input_port)
    output_ports.add(c.upstream_output_port)

input_ports = sorted(input_ports)
output_ports = sorted(output_ports)
```